code_challenges/code_wars/count_duplicates.py

```diff
@@ -0,0 +1,12 @@
+# https://www.codewars.com/kata/54bf1c2cd5b56cc47f0007a1
+
+
+def count_duplicates(characters):
+    freq_of_dupes = {}
+    for char in characters.lower():
+        freq_of_dupes[char] = freq_of_dupes.get(char, 0) + 1
+    duplicate_chars = [char for char, count in freq_of_dupes.items() if count > 1]
+    if len(duplicate_chars) == 0:
+        return 0
+
+    return len(duplicate_chars)
```

Comment:
Whats the difference between
```python
    if len(duplicate_chars) == 0:
        return 0

    return len(duplicate_chars)
```
and
```python
    return len(duplicate_chars)
```
 ?

Reply:
it would be computing len() again but should I not really be too concerned about O(1)?

Reply:
Think maybe I wasn't clear with the question.

On line 9 - 10, 
```python
if len(duplicate_chars) == 0:
    return 0
``` 
if the length of dups is 0 you return 0 (the length).

Then if the length isn't zero we move on to line 12 and still return the length.  So why are you checking the length at all to see if it's zero if you always return the length in any case?

Lines 9-10 can be removed and the function will behave the exact same way.

----
Unrelated to my observation at all,
> should I not really be too concerned about O(1)

If you can avoid an operation without making your code harder to read you should avoid it, but in general you are correct.  You shouldn't be too concerned about O(1) additions to solutions.

Reply:
I tried really hard to figure out why I put it there but I cannot come up with anything so it was definitely in error.

would there even be a benefit to add a zero check **before** entering the frequency mapping:

```
def count_duplicates(characters):
    freq_of_dupes = {}
    if len(characters) == 0:
      return 0
    for char in characters.lower():
        freq_of_dupes[char] = freq_of_dupes.get(char, 0) + 1
    duplicate_chars = [char for char, count in freq_of_dupes.items() if count > 1]
    return len(duplicate_chars)
```

Reply:
> would there even be a benefit to add a zero check before entering the frequency mapping:

Not really. It would be fine, to break out early, but also the for loop, list comp and len would all be nearly instant on an empty string, so not much saving.